### PR TITLE
Feature/minkowski example

### PR DESCRIPTION
**Pull request by elenanst**
I have converted "distance_minkowski_modular.py" with a meta example and a cookbook page.

### CODE REVIEW

**Comment by vigsterkr** [comment]
@elenanst please check the errors you are getting from CI and try to fix them. a good advise: try to run the tests before sending a PR

**Comment by karlnapf** [comment]
We actually dont need more cookbook pages for distance. Just the meta example is fine

File: doc/cookbook/source/examples/distance/minkowski.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
+====================
+Minkowski Distance
+====================
+
+The Minkowski distance is a general class of distances for a :math:`R^n` feature space and is defined as
```
Comment (karlnapf) [inline doc/cookbook/source/examples/distance/minkowski.rst]: use "d" for dimension please

File: doc/cookbook/source/examples/distance/minkowski.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+.. math::
+
+    \big(\sum_{i=1}^{n} |{\bf x_i} - {\bf x'_i}|^k \big)^{\frac{1}{k}}
```
Comment (karlnapf) [inline doc/cookbook/source/examples/distance/minkowski.rst]: use `\left ( ` and \`\right )`

File: doc/cookbook/source/examples/distance/minkowski.rst [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+-------
+Example
+-------
+
+Imagine we have files with data. We create CDenseFeatures (here 64 bit floats aka RealFeatures) as
+
+.. sgexample:: minkowski.sg:create_features
+
+We create an instance of :sgclass:`CMinkowskiMetric` by passing it :sgclass:`CDenseFeatures`. We choose the order :math:`k` to be 3.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/distance/minkowski.rst]: please dont put 3 in the text. It is already in the code.
Just say "We choose the order k"

**Comment by elenanst** [comment]
Running make && make test on my machine reports: 

99% tests passed, 1 tests failed out of 445

Total Test time (real) =  40.41 sec

The following tests FAILED:
	 96 - unit-UAIFileTest (Failed)
Errors while running CTest
Makefile:61: recipe for target 'test' failed
make: *** [test] Error 8

But this is also happening on my up-to-date develop branch, so I'm assuming it's not a problem caused by my committed files. Same for my other PR  https://github.com/shogun-toolbox/shogun/pull/3728

**Pull request closed without merging**